test/sagemaker_tests/huggingface/inference/integration/__init__.py

```diff
@@ -28,3 +32,37 @@
 DEFAULT_TIMEOUT = 20
 
 RESOURCE_PATH = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "resources"))
+
+
+class NoLogStreamFoundError(Exception):
+    pass
+
+
+class SageMakerEndpointFailure(Exception):
+    pass
+
+
+def dump_logs_from_cloudwatch(e):
+    """
+    Function to dump logs from cloudwatch during error handling
+    """
+    endpoint_regex = re.compile(r"Error hosting endpoint ((\w|-)+):")
+    endpoint_match = endpoint_regex.search(str(e))
+    if endpoint_match:
+        logs_client = boto3.client('logs', region_name='us-west-2')
+        endpoint = endpoint_match.group(1)
+        log_group_name = f"/aws/sagemaker/Endpoints/{endpoint}"
+        log_stream_resp = logs_client.describe_log_streams(logGroupName=log_group_name)
+        all_traffic_log_stream = ""
+        for log_stream in log_stream_resp.get('logStreams', []):
+            log_stream_name = log_stream.get('logStreamName')
+            # If we have AllTraffic log stream, just use that
```

Comment:
nit: it would be good if we can add the expected format for logstream name in the comment